### PR TITLE
rtl8852bu: init at f6aaa3c

### PR DESCRIPTION
**Pull request by lonyelon**
## Description of changes

Add the RTL8852BU driver for some devices that need it. Source at https://github.com/morrownr/rtl8852bu.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lonyelon** [comment]
@rjpcasalino As we said in the #279702 I made a new PR for 8852bu. This is building on my system but I have no way to test if it works, can you confirm it for me please?

File: pkgs/os-specific/linux/rtl8852bu/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  '';
+
+  postInstall = ''
+    nuke-refs $out/lib/modules/*/kernel/net/wireless/*.ko
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "Driver for Realtek rtl8852au and rtl8832bu chipsets, provides the 8852au mod";
```
Comment (rjpcasalino) [inline pkgs/os-specific/linux/rtl8852bu/default.nix:45]: provides the 8852bu?

**Review by rjpcasalino — APPROVED** [review]

**Comment by rjpcasalino** [comment]
I've been using this change essentially for a month and it works well-thanks for adding it officially

**Comment by rjpcasalino** [comment]
Result of `nixpkgs-review pr 306247` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_19_hardened.rtl8852bu (linuxKernel.packages.linux_4_19_hardened.rtl8852bu)</li>
    <li>linuxPackages_5_10_hardened.rtl8852bu (linuxKernel.packages.linux_5_10_hardened.rtl8852bu)</li>
    <li>linuxPackages_5_15_hardened.rtl8852bu (linuxKernel.packages.linux_5_15_hardened.rtl8852bu)</li>
    <li>linuxPackages_5_4_hardened.rtl8852bu (linuxKernel.packages.linux_5_4_hardened.rtl8852bu)</li>
    <li>linuxPackages_6_1_hardened.rtl8852bu (linuxKernel.packages.linux_6_1_hardened.rtl8852bu)</li>
    <li>linuxPackages_hardened.rtl8852bu (linuxPackages_6_6_hardened.rtl8852bu)</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.rtl8852bu</li>
    <li>linuxKernel.packages.linux_5_10.rtl8852bu</li>
    <li>linuxKernel.packages.linux_5_15.rtl8852bu</li>
    <li>linuxKernel.packages.linux_5_4.rtl8852bu</li>
    <li>linuxKernel.packages.linux_6_1.rtl8852bu</li>
    <li>linuxPackages.rtl8852bu (linuxKernel.packages.linux_6_6.rtl8852bu)</li>
    <li>linuxPackages_latest.rtl8852bu (linuxKernel.packages.linux_6_8.rtl8852bu)</li>
    <li>linuxPackages_latest-libre.rtl8852bu (linuxKernel.packages.linux_latest_libre.rtl8852bu)</li>
    <li>linuxPackages-libre.rtl8852bu (linuxKernel.packages.linux_libre.rtl8852bu)</li>
    <li>linuxPackages_lqx.rtl8852bu (linuxKernel.packages.linux_lqx.rtl8852bu)</li>
    <li>linuxPackages_xanmod.rtl8852bu (linuxKernel.packages.linux_xanmod.rtl8852bu)</li>
    <li>linuxPackages_xanmod_latest.rtl8852bu (linuxKernel.packages.linux_xanmod_latest.rtl8852bu ,linuxPackages_xanmod_stable.rtl8852bu)</li>
    <li>linuxPackages_zen.rtl8852bu (linuxKernel.packages.linux_zen.rtl8852bu)</li>
  </ul>
</details>

**Review by rjpcasalino — COMMENTED** [review]
seeing some errors but unsure if it should block us

**Comment by lonyelon** [comment]
> Result of `nixpkgs-review pr 306247` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 6 packages failed to build:
> 13 packages built:

Well, the other PR (#279702) had more fails so I see this as an improvement : )

**Review by rjpcasalino — APPROVED** [review]

**Comment by lonyelon** [comment]
Well, that was exceptionally fast lol

**Comment by kirillrdy** [comment]
well, this PR is very similar to previous one :-)

**Comment by rjpcasalino** [comment]
@lonyelon sorry to bother you on an old thread - I see you have kept this package up to date but no matter which version I use I get the same error (missing drv_types.h header file?)

```
       last 25 log lines:
       >    17 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/891qcsv4m0plkrqcpgk3p1qxbqbs97vy-linux-6.14.3-dev/lib/modules/6.14.3/source/scripts/Makefile.build:207: os_dep/linux/ioctl_cfg80211.o] Error 1
       > os_dep/linux/rtw_cfgvendor.c:16:10: fatal error: drv_types.h: No such file or directory
       >    16 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/891qcsv4m0plkrqcpgk3p1qxbqbs97vy-linux-6.14.3-dev/lib/modules/6.14.3/source/scripts/Makefile.build:207: os_dep/linux/rtw_cfgvendor.o] Error 1
       > os_dep/linux/os_ch_utils.c:17:10: fatal error: drv_types.h: No such file or directory
       >    17 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/891qcsv4m0plkrqcpgk3p1qxbqbs97vy-linux-6.14.3-dev/lib/modules/6.14.3/source/scripts/Makefile.build:207: os_dep/linux/os_ch_utils.o] Error 1
       > os_dep/linux/wifi_regd.c:16:10: fatal error: drv_types.h: No such file or directory
       >    16 | #include <drv_types.h>

```
Wondering if you know of a quick fix?

**Pull request merged**